### PR TITLE
Style: Added inline code styling 

### LINKED ISSUE #5934: Create `inline code block` styling
Create a inline code `<code/>` styling using the new [figma design (pagelink)](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=696-15704&mode=dev). 

This isn't a new React component, just a globally styled HTML element we want to illustrate. The primary use case for this styling is \`code\` tags within rendered markdown/MDX files.

> [Read more about the Node.js Website redesign](https://github.com/nodejs/nodejs.org/discussions/5131)

---

### Before You Start...

- Use the [Contributor's Guide](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) to learn how to [setup your environment](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#getting-started), run [available commands](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#cli-commands), and construct [proper commits](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#commit-guidelines).
- Review the [technologies used](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#technologies-used-in-the-website) when building the site, and [why they were chosen](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#remarks-on-technologies-used).

---

### Details

image

image

❓ [Not sure where to start breaking down the figma? Read this short guide](https://github.com/nodejs/nodejs.org/discussions/5131#discussioncomment-6848378)

- Add these code block styles to the existing [storybook story file `components/__design__/text.stories.tsx`](https://github.com/nodejs/nodejs.org/blob/main/components/__design__/text.stories.tsx)which will excercise each of the component's states.
- Use the new tailwind color and typography system to assign properties. For example, Light `background` uses Neutral/100 per the figma.

There are 2 states to capture within styles and stories:

- Light
- Dark

### Open Questions

Could we make this an `InlineCodeblock` component and then add it to an MDX provider? That might isolate markdown-driven use cases more, but at the cost of increased complexity. For now, let's show what an `<code/>` tag renders like. Perhaps what it looks like within `<p>` tag only?

### PR DESCRIPTION
**Pull request by akhilender-bongirwar**
## Description
In this commit, I've implemented inline code block styling according to the Figma design.

Changes Made:
- Created `styles/new/base.css` and added styling for code blocks in light mode.
- Created `styles/new/layout/dark-theme.css` and added styling for code blocks in dark mode.
- Created `variables.css` to declare variables.

For these changes, I followed the existing file structure in the `styles/old` folder.

## Validation
Fixes #5934 

### Check List

- [X] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [X] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [X] I have run `npx turbo format` to ensure the code follows the style guide.
- [X] I have run `npx turbo test` to check if all tests are passing.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by ovflowd — CHANGES_REQUESTED** [review]
Please read our Collaborator Guidelines. We don't do manual CSS here as we use Tailwind.

Tailwind has already all the font sizes, colours, spacings, font sizes, spacing and etc.

You might only need to update `tailwind` config by adding the font families for the inline code.

You might want to follow Tailwind's guide for adding new font-families (ie, you need to update `.storybook/preview.tsx` with the IBM Plex Mono font and also `tailwind.config`

Not to mention you forgot to add the Storybooks here.

**Comment by akhilender-bongirwar** [comment]
@ovflowd, my apologies for not having read the Collaborator Guidelines previously. I have now adhered to the guidelines by using only Tailwind CSS classes for styling. Additionally, could you please provide guidance on how to add storybooks here?

**Comment by ovflowd** [comment]
> Additionally, could you please provide guidance on how to add storybooks here?

You mentioned you read the Collaborator Guides, if you did, you would know how to. It's all written there. (I know it's a lot of content to read, so no worries. Take your time)

Also the issue itself probably states what you should create for the Storybook.

File: styles/new/base.css [hunk 1]
```diff
@@ -4,3 +4,10 @@ a:visited {
   @apply text-green-600
    dark:text-green-400;
 }
+
+code {
```
Comment (ovflowd) [inline styles/new/base.css:8]:
```css
code {
  @apply font-mono 
    text-base 
    font-semibold 
    text-neutral-900;
}
```

File: styles/new/layout/dark-theme.css
Comment (ovflowd) [inline styles/new/layout/dark-theme.css:1]: That's not the way of doing dark theming. Read https://tailwindcss.com/docs/dark-mode

You simply add `dark:` as a prefix for the dark things.

Also, you don't need to define again `text-base`, `font-semibold` and all of the other things already set on the base styles. You only need to add on base.css `dar:text-white` (I'd recommend you to give a better read on how CSS works).

Also, inline code should have a background colour and a border-radius and I'm not seeing here.

File: tailwind.config.ts [hunk 2]
```diff
@@ -111,6 +111,7 @@ export default {
     },
     fontFamily: {
       'open-sans': ['var(--font-open-sans)'],
+      mono: ['var(--font-ibm-plex-mono)'],
```
Comment (ovflowd) [inline tailwind.config.ts]: This should be called 'ibm-plex-mono'

```suggestion
      'ibm-plex-mono': ['var(--font-ibm-plex-mono)'],
```

File: styles/new/base.css [hunk 3]
```diff
@@ -4,3 +4,15 @@ a:visited {
...
+  @apply rounded-[4px] 
+    bg-neutral-100
+    px-[4px]
+    font-ibm-plex-mono
+    text-base 
+    font-semibold
+    text-neutral-900
+   dark:bg-neutral-900
```
Comment (ovflowd) [inline styles/new/base.css]: The `dark` styles need to be indented (extra space) too 👀

File: components/__design__/text.stories.tsx [hunk 4]
```diff
@@ -21,4 +21,12 @@ export const Anchor: StoryObj = {
   ),
 };
 
+export const Code: StoryObj = {
```
Comment (ovflowd) [inline components/__design__/text.stories.tsx]:
```suggestion
export const InlineCode: StoryObj = {
```

**Review by ovflowd — APPROVED** [review]
Everything seems alright! Thanks for making this contribution and for making the updates based on feedback! Appreciate it!

I'm just verifying the Stories now, but afaik it looks alright!

**Comment by ovflowd** [comment]
> NextFontError: Preload is enabled but no subsets were specified for font `IBM Plex Mono`. Please specify subsets or disable preloading if your intended subset can't be preloaded.
Available subsets: `cyrillic`, `cyrillic-ext`, `latin`, `latin-ext`, `vietnamese`

FYI apparently there is missing a subset setting on the font https://nextjs.org/docs/messages/google-fonts-missing-subsets can you set the subset as `latin`?

**Comment by akhilender-bongirwar** [comment]
> > NextFontError: Preload is enabled but no subsets were specified for font `IBM Plex Mono`. Please specify subsets or disable preloading if your intended subset can't be preloaded.
> > Available subsets: `cyrillic`, `cyrillic-ext`, `latin`, `latin-ext`, `vietnamese`
> 
> FYI apparently there is missing a subset setting on the font https://nextjs.org/docs/messages/google-fonts-missing-subsets can you set the subset as `latin`?

Should I even add display: 'fallback' with subsets:['latin'] ? // @ovflowd

**Review by ovflowd — APPROVED** [review]

**Review by AugustinMauroy — APPROVED** [review]
SGTM ! Thanks for your first contribution 🎉

**Comment by akhilender-bongirwar** [comment]
> Everything seems alright! Thanks for making this contribution and for making the updates based on feedback! Appreciate it!
> 
> I'm just verifying the Stories now, but afaik it looks alright!

@ovflowd, I wanted to express my gratitude for your support. Initially, I felt overwhelmed working on a large repository, but I'm actively learning how to collaborate effectively within well-structured projects. Your guidance and assistance have been invaluable, and I appreciate it greatly. Thank you for your patience and encouragement in my learning journey.

**Comment by ovflowd** [comment]
> > Everything seems alright! Thanks for making this contribution and for making the updates based on feedback! Appreciate it!
> > I'm just verifying the Stories now, but afaik it looks alright!
> 
> @ovflowd, I wanted to express my gratitude for your support. Initially, I felt overwhelmed working on a large repository, but I'm actively learning how to collaborate effectively within well-structured projects. Your guidance and assistance have been invaluable, and I appreciate it greatly. Thank you for your patience and encouragement in my learning journey.

It can be overwhelming, yes. Hopefully, our docs were good enough and our support too 🙇

**Pull request merged**